From 1d9ac57c8318a72b264eabf9d84f42097ab8c438 Mon Sep 17 00:00:00 2001
From: Wei-Chiu Chuang <weichiu@cloudera.com>
Date: Wed, 25 May 2016 15:18:24 -0700
Subject: [PATCH 1582/2748] CLOUDERA-BUILD. CDH-40786 Addendum patch for
 HDFS-10360 (DataNode may format directory and
 lose blocks if current/VERSION is missing)

Change-Id: Ie7b6a52530866a1455626875c34ea7e3b3314ec8
---
 .../server/datanode/BlockPoolSliceStorage.java     |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockPoolSliceStorage.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockPoolSliceStorage.java
index a6f1632..25d4bbd 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockPoolSliceStorage.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockPoolSliceStorage.java
@@ -151,7 +151,7 @@ private StorageDirectory loadStorageDirectory(NamespaceInfo nsInfo,
           throws IOException {
     StorageDirectory sd = new StorageDirectory(dataDir, null, true);
     try {
-      StorageState curState = sd.analyzeStorage(startOpt, this);
+      StorageState curState = sd.analyzeStorage(startOpt, this, true);
       // sd is locked but not opened
       switch (curState) {
       case NORMAL:
-- 
1.7.9.5

